sia-gpu-miner.c

```diff
@@ -302,10 +313,33 @@ void printPlatformsAndDevices() {
 			ret = clGetDeviceInfo(deviceids[j], CL_DEVICE_NAME, 80, str, NULL);
 			if (ret != CL_SUCCESS) {
 				printf("\tError while getting device info.\n");
-				free(deviceids);
-				continue;
-			}
-			printf("\tDevice %d: %s\n", j, str);
+            } else {
+                printf("\tDevice %d: %s\n", j, str);
+            }
+            
+            cl_bool isLitlle = CL_TRUE;
+            ret = clGetDeviceInfo(deviceids[j], CL_DEVICE_ENDIAN_LITTLE, sizeof(isLitlle), &isLitlle, NULL);
+            if (ret != CL_SUCCESS) {
+                printf("\tError while getting device CL_DEVICE_ENDIAN_LITTLE.\n");
+            } else {
+                printf("\tDevice %d: isLitlle: %d\n", j, isLitlle);
+            }
```

Comment:
could be more descriptive here. How about:
```c
else if (isLittle) {
    printf("\t\tEndianness: Little-endian\n");
} else {
    printf("\t\tEndianness: Big-endian\n");
}
```
also, please ensure that your new code is properly indented.